libraries/redshop/helper/tax.php

```diff
@@ -114,7 +114,7 @@ public static function getVatRates($productId = 0, $userId = 0)
                     . ' OR ' . $db->qn('tr.tax_country') . ' IS NULL))'
                 )
                 ->where('tr.tax_group_id = ' . (int)$taxGroup)
-	            ->where($db->qn('tsx.shopper_group_id') . ' = ' . $db->q($userArr['rs_user_shopperGroup']))
+	            ->where($db->qn('tsx.shopper_group_id') . ' = ' . $db->q(isset($userData->shopper_group_id) ? $userData->shopper_group_id : $userArr['rs_user_shopperGroup']))
```

Comment:
@hoang-tran-redweb better that using a variable get the result and place it here. Avoid to use complex nested code.

Reply:
@anhredweb I fixed. please check again.